### PR TITLE
Readme rework for 0.4 release

### PR DESCRIPTION
**Pull request by markurtz**
- Refactoring structure based on new standards
- Adding in highlights, tutorials, and infographic images
- Resetting docs root to compile readme

### CODE REVIEW

File: docs/source/code.md [hunk 1]
```diff
@@ -125,7 +89,7 @@ The integration is done using the `ScheduledModifierManager` class which can be
 This class handles modifying the TensorFlow graph for the desired algorithms.
 With this setup, the training process can then be modified as desired to sparsify the model.
 
-#### Estimator-Based pipelines
+### Estimator-Based pipelines
```
Comment (jeanniefinks) [inline docs/source/code.md:92]:
```suggestion
### Estimator-Based Pipelines
```

File: docs/source/code.md [hunk 2]
```diff
@@ -142,7 +106,7 @@ manager.modify_estimator(estimator, steps_per_epoch=num_train_batches)
 # Normal estimator training code...
 ```
 
-#### Session-Based pipelines
+### Session-Based pipelines
```
Comment (jeanniefinks) [inline docs/source/code.md:109]:
```suggestion
### Session-Based Pipelines
```

File: docs/source/installation.md [hunk 3]
```diff
@@ -14,15 +14,57 @@ See the License for the specific language governing permissions and
...
 ```bash
 pip install sparseml
 ```
 
-Then if you would like to explore any of the [scripts](https://github.com/neuralmagic/sparseml/tree/main/scripts), [notebooks](https://github.com/neuralmagic/sparseml/tree/main/notebooks), or [integrations](https://github.com/neuralmagic/sparseml/tree/main/integrations) clone the repository and install any additional dependencies as required.
+## Supported Framework Versions
+
+The currently supported framework versions are:
+
+- PyTorch supported versions: `>= 1.1.0, < 1.8.0`
```
Comment (jeanniefinks) [inline docs/source/installation.md:33]:
```suggestion
- PyTorch-supported versions: `>= 1.1.0, < 1.8.0`
```

File: docs/source/installation.md [hunk 4]
```diff
@@ -14,15 +14,57 @@ See the License for the specific language governing permissions and
...
+- Keras supported versions: `2.3.0-tf` (through the TensorFlow `2.2` package; as of Feb 1st, 2021, `keras2onnx` has
```
Comment (jeanniefinks) [inline docs/source/installation.md:34]:
```suggestion
- Keras-supported versions: `2.3.0-tf` (through the TensorFlow `2.2` package; as of Feb 1st, 2021, `keras2onnx` has
```

File: docs/source/installation.md [hunk 5]
```diff
@@ -14,15 +14,57 @@ See the License for the specific language governing permissions and
...
+not been tested for TensorFlow >= `2.3`). 
+- TensorFlow V1 supported versions: >= `1.8.0` (TensorFlow >= `2.X` is not currently supported)
```
Comment (jeanniefinks) [inline docs/source/installation.md:36]:
```suggestion
- TensorFlow V1-supported versions: >= `1.8.0` (TensorFlow >= `2.X` is not currently supported)
```

File: docs/source/code.md [hunk 6]
```diff
@@ -14,73 +14,37 @@ See the License for the specific language governing permissions and
...
 Inside are APIs designed to make model sparsification as easy as possible by integrating seamlessly into PyTorch training pipelines.
 
-The integration is done using the `ScheduledOptimizer` class. 
-It is intended to wrap your current optimizer and its step function. 
-The step function then calls into the `ScheduledModifierManager` class which can be created from a recipe file. 
+The step function then calls into the `ScheduledModifierManager` class which can be created from a recipe file.
+The `modify()` function wraps an optimizer or optimizer like object (contains a step function) to override the step invocation.
 With this setup, the training process can then be modified as desired to sparsify the model.
 
 To enable all of this, the integration code you'll need to write is only a handful of lines:
```
Comment (jeanniefinks) [inline docs/source/code.md:28]:
```suggestion
To enable all of this, the integration code is accomplished by writing a handful of lines:
```

Comment (jeanniefinks) [reply docs/source/code.md:28]: Also, commit that suggestion to the identical Keras line below?

**Review by jeanniefinks — CHANGES_REQUESTED** [review]
simple grammatical changes

File: README.md [hunk 7]
```diff
@@ -14,32 +14,33 @@ See the License for the specific language governing permissions and
 limitations under the License.
 -->
 
-# ![icon for SparseMl](https://raw.githubusercontent.com/neuralmagic/sparseml/main/docs/source/icon-sparseml.png) SparseML
+<h1><img src="https://raw.githubusercontent.com/neuralmagic/sparseml/main/docs/source/icon-sparseml.png" />&nbsp;&nbsp;SparseML</h1>
```
Comment (jeanniefinks) [inline README.md:17]:
```suggestion
<h1><img alt="project icon" src="https://raw.githubusercontent.com/neuralmagic/sparseml/main/docs/source/icon-sparseml.png" />&nbsp;&nbsp;SparseML</h1>
```

File: README.md [hunk 8]
```diff
@@ -52,345 +53,153 @@ limitations under the License.
...
+The [GitHub repository](https://github.com/neuralmagic/sparseml) contains integrations within the PyTorch, Keras, and TensorFlow V1 ecosystems, allowing for seamless model sparsification.
 
-Session-based pipelines need a little bit more as compared to estimator-based pipelines; however,
-it is still designed to require only a few lines of code for integration.
-After graph creation, the manager's `create_ops` method must be called.
-This will modify the graph as needed for the algorithms and return modifying ops and extras.
-After creating the session and training normally, call into `session.run` with the modifying ops after each step.
-Modifying extras contain objects such as tensorboard summaries of the modifiers to be used if desired.
-Finally, once completed, `complete_graph` must be called to remove the modifying ops for saving and export.
+<img src="https://docs.neuralmagic.com/docs/source/infographics/sparseml.png" width="960px" />
```
Comment (jeanniefinks) [inline README.md:61]:
```suggestion
<img alt="SpaseML Flow" src="https://docs.neuralmagic.com/docs/source/infographics/sparseml.png" width="960px" />
```

**Review by jeanniefinks — APPROVED** [review]

**Review by NMsasa — APPROVED** [review]

**Review by bfineran — APPROVED** [review]

**Pull request merged**